### ISSUE TITLE
Upgrade or init only updates ck8s version in default config

### ISSUE THREAD

**Issue by Elias-elastisys** [elastisys/compliantkubernetes-apps#1539]
**Describe the bug**
When running `bin/ck8s upgrade <version> prepare`, (or just init), it only updates the `ck8sVersion` in `defaults/common-config.yaml`, so if you have that value defined in the override `common-config.yaml`, the upgrade process fails when trying to apply the update.

**To Reproduce**
1. Add an override to ck8sVersion` in `common-config.yaml`
2. Run `bin/ck8s upgrade <version> prepare`
3. Run `bin/ck8s upgrade <version> apply`

**Expected behavior**
Prepare/init should update config correctly to allow for update.

**Version (add all relevant versions):**
 - Compliant kubernetes apps version v0.29.0

**Additional context**
When trying to upgrade from v0.29.0 to v0.29.1.

**Comment by Xartos** [comment]
This is by design.
If you've set the version in the override `common-config.yaml` then it's probably for a reason and thus shouldn't be automatically updated because this might break whatever change you did for the last version.

Or do you have any other reason why this would be nice to have?

**Comment by Elias-elastisys** [comment]
Okay understandable. 

No I just felt as if it was silently breaking sort of. I was working on a shared config and it was not me that had added the version override, so I missed it. And the prepare step seemed to complete successfully. To me it looked like the upgrade process was broken. 

Maybe if its possible to add a warning or message when preparing that the upgrade won't work. But this is not a huge issue.

**Comment by Xartos** [comment]
Aah, yea if it silently breaks then it's a problem. That we should probably fix

**Comment by Elias-elastisys** [comment]
Yeah the prepare step seems to complete successfully, and it only fails on apply with an version mismatch error.
